sessionctx/variable/tidb_vars.go

```diff
@@ -195,6 +195,9 @@ const (
 	// tidb_opt_insubquery_to_join_and_agg is used to enable/disable the optimizer rule of rewriting IN subquery.
 	TiDBOptInSubqToJoinAndAgg = "tidb_opt_insubq_to_join_and_agg"
 
+	// tidb_opt_always_prefer_index is used to enable/disable the optimizer to always prefer index scan over table scan, ignoring their costs.
+	TiDBOptAlwaysPreferIndex = "tidb_opt_always_prefer_index"
```

Comment:
how about:

```suggestion
	TiDBOptPreferRangeScan = "tidb_opt_prefer_range_scan"
```

Reply:
- for clustered primary key: table range scan is preferred
- for secondary index: index range scan + table rowid scan is preferred